src/main/cpp/src/cast_string_to_float.cu

```diff
@@ -641,10 +641,9 @@ std::unique_ptr<column> string_to_float(data_type dtype,
                                         rmm::cuda_stream_view stream,
                                         rmm::mr::device_memory_resource* mr)
 {
-  if ((string_col.size() == 0) ||
-      (dtype != data_type{type_id::FLOAT32} && dtype != data_type{type_id::FLOAT64})) {
-    return std::make_unique<column>();
-  }
+  CUDF_EXPECTS(dtype == data_type{type_id::FLOAT32} || dtype == data_type{type_id::FLOAT64},
+               "invalid float data type");
+  if (string_col.size() == 0) { return std::make_unique<column>(); }
```

Comment:
Probably empty column of floating point type?

Reply:
Good point, I should. Thanks.